impl/maven-impl/src/main/java/org/apache/maven/impl/DefaultModelXmlFactory.java

```diff
@@ -48,8 +48,37 @@
 @Singleton
 public class DefaultModelXmlFactory implements ModelXmlFactory {
     @Override
+    @Nonnull
     public Model read(@Nonnull XmlReaderRequest request) throws XmlReaderException {
         nonNull(request, "request");
+        Model model = doRead(request);
+        if (isModelVersionGreaterThan400(model)
+                && !model.getNamespaceUri().startsWith("http://maven.apache.org/POM/")) {
```

Comment:
Here you can just check the namespace for full equality rather than using startsWith to check only the prefix.

Reply:
Well, currently the code needs to accept both `http://maven.apache.org/POM/4.0.0` and `http://maven.apache.org/POM/4.1.0`, so I did not want enforce too much right now, whatever the outcome of that other discussion is.